components/Common/Search/States/WithAllResults.tsx

```diff
@@ -0,0 +1,33 @@
+import type { Results } from '@orama/orama';
+import NextLink from 'next/link';
+import { useTranslations } from 'next-intl';
+import type { FC } from 'react';
+
+import type { SearchDoc } from '@/components/Common/Search/States/WithSearchBox';
+
+import styles from './index.module.css';
+
+type SearchResults = Results<SearchDoc>;
+
+type SeeAllProps = {
+  searchResults: SearchResults;
+  searchTerm: string;
+  selectedFacetName: string;
+};
+
+export const WithAllResults: FC<SeeAllProps> = props => {
+  const t = useTranslations();
+  const resultsCount = props.searchResults?.count?.toLocaleString('en') ?? 0;
+
+  const sanitizedSearchTerm = encodeURIComponent(props.searchTerm);
+  const sanitizedFacetName = encodeURIComponent(props.selectedFacetName);
+  const allResultsURL = `/en/search?q=${sanitizedSearchTerm}&section=${sanitizedFacetName}`;
```

Comment:
Really good catch. Manual /en/ shouldn't be done, we should use our Link component from "components" folder that does the localization for you. Also, could you use `URLSearchParams` instead of manually writing the query params?